### PR TITLE
Update the runtime code where iOS/MacCatalyst logic is currently used to benefit from the updated analyzer behavior

### PR DESCRIPTION
**Pull request by buyaa-n**
For resolving https://github.com/dotnet/runtime/issues/53084 we [planned the following steps]( https://github.com/dotnet/runtime/issues/53084#issuecomment-878722932):

1. [x] Update the runtime to have `OperatingSystem.IsIOS` return true when on MacCatalyst, and apply the minimal guard attributes (.NET 6, Preview 7)
2. [x] Update the analyzer per the proposal (.NET 6, RC1)
3. [X]  :heavy_check_mark:  Update the runtime code where iOS/MacCatalyst logic is currently used to benefit from the updated analyzer behavior (.NET 6, RC1)
~~4. [ ] Optionally make another change to the runtime to model the macOS/OSX alias through the guard attributes (not for 6.0)~~

As you can see steps 1 and 2 are completed and this PR is for step 3 `Update the runtime code where iOS/MacCatalyst logic is currently used to benefit from the updated analyzer behavior`

The PR includes the following updates:
- Remove duplicate `!OperatingSystem.IsMacCatalyst()` when there is `!OperatingSystem.IsIOS()` guard also used
- Remove `[UnsupportedOSPlatformGuard("maccatalyst")]` guard attribute when there is `[UnsupportedOSPlatformGuard("ios")]` attribute applied
- Remove `[UnsupportedOSPlatform("maccatalyst")]` attribute when there is `[UnsupportedOSPlatform("ios")]` attribute applied

Side notes:
- There were no any `[SupportedOSPlatform("maccatalyst")]` nor `[SupportedOSPlatformGuard("maccatalyst")]` 
- There were no any `[UnsupportedOSPlatform("maccatalyst")]` applied without `[UnsupportedOSPlatform("ios")]`, same for `[UnsupportedOSPlatformGuard("maccatalyst")]`
- Not sure if there is an edge case where we want to add `[SupportedOSPlatform("maccatalyst")]` for existing  `[UnsupportedOSPlatform("ios")]`, to express the API is only unsupported on `ios`, the same question for guard attributes/methods. If there is such case this PR doesn't include that

CC @jeffhandley @marek-safar @terrajobst @filipnavara

### CODE REVIEW

File: src/libraries/System.Console/src/System/Console.cs [hunk 1]
```diff
@@ -38,8 +38,7 @@ public static class Console
         [UnsupportedOSPlatform("android")]
         [UnsupportedOSPlatform("browser")]
         [UnsupportedOSPlatform("ios")]
-        [UnsupportedOSPlatform("maccatalyst")]
-        [UnsupportedOSPlatform("tvos")]
+         [UnsupportedOSPlatform("tvos")]
```
Comment (filipnavara) [inline src/libraries/System.Console/src/System/Console.cs:41]: nit: Indentation

**Review by filipnavara — APPROVED** [review]
LGTM (as long as CI passes). Thanks!

**Comment by filipnavara** [comment]
The failure on MacCatalyst CI is tracked by https://github.com/dotnet/runtime/issues/57010.

**Review by jeffhandley — APPROVED** [review]

**Comment by buyaa-n** [comment]
> The failure on MacCatalyst CI is tracked by #57010.

Thanks, seems all failures related to that, merging

**Pull request merged**